app/main.py

```diff
@@ -1,3 +1,55 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: int | float) -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        return Distance(self.km + other)
+
+    def __iadd__(self, other: int | float) -> "Distance":
```

Comment:
The `__iadd__` method should accept `other` as `Union[int, float, Distance]` to match the logic inside the method. This ensures that the method can handle both `Distance` objects and numeric values.